Set starting directory for QT file dialog

JacksonRG opened the pull request:
Default folder for file imports wasn't being updated. Always starting in the same folder.

Fixed by using the `setDirectory` function rather than the `getOpenFileUrls`'s `directory` argument which some file dialogs interpret as a search term rather than Folder Path.

Commit:
Set starting directory for QT file dialog

Use `setDirectory` to set the file picker's starting folder

JacksonRG commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Import/Export Folders in Settings](https://trello.com/c/ABZbyHRd/48-import-export-folders-in-settings)

jonoomph committed:
Fixing default path of imported files to use localFile path

jonoomph committed:
Merge branch 'develop' into locations-appimage-fix

JacksonRG commented:
@jonoomph Figured out why the setting wasn't changing on *initial save* of a new project.

I cutting off the filename only IF `os.path.isfile()` Which returns true for *existing* file names. Anyway, now I remove the file name if `not os.path.isdir()`

Commit:
Ensure all actions use and update location settings

- Remove import_path and export_path from default settings.
- Update settings on video export
- Don't default to most recent project folder
- Update folder on first save of a new project.

jonoomph commented on src/classes/project_data.py:
```diff
@@ -947,7 +942,8 @@ def check_if_paths_are_valid(self):
                     # Found file, update path
                     file["path"] = path
-                    get_app().updates.update_untracked(["import_path"], os.path.dirname(path))
+                    settings
```
What is this?

JacksonRG replied:
:grimacing: an unfinished change. It was going to be `settings.setDefaultPath...` fixing

jonoomph commented on src/classes/project_data.py:
```diff
@@ -947,7 +942,8 @@ def check_if_paths_are_valid(self):
                     file["path"] = path
-                    get_app().updates.update_untracked(["import_path"], os.path.dirname(path))
+                    settings
+                    settings.setDefaultPath(settings.actionType.LOAD, os.path.dirname(path))
```
Should this use `IMPORT` instead of `LOAD` - just based on the previous behavior?

jonoomph commented on src/classes/settings.py:
```diff
@@ -141,10 +141,12 @@ def setDefaultPath(self, action: pathType, recent_path: str):
-        if os.path.isfile(recent_path):
-            recent_path=os.path.dirname(recent_path)
+        if not os.path.isdir(recent_path):
+            # log.info("File path. reducing to directory")
```
Fix commented out debug lines - seems useful to keep this logged (file path => folder path)

jonoomph commented on src/windows/export.py:
```diff
@@ -790,7 +790,8 @@ def titlestring(sec, fps, mess):
         # Store updated export folder path in project file
-        get_app().updates.update_untracked(["export_path"], os.path.dirname(export_file_path))
+        settings = get_app().get_settings()
+        settings.setDefaultPath(settings.actionType.EXPORT, os.path.dirname(export_file_path))
```
Doesn't the setDefaultPath method already handle file_paths?  i.e. by reducing them to the folder path?

JacksonRG replied:
It does. I can take out the `dirname` for simplicity

jonoomph commented on src/windows/main_window.py:
```diff
@@ -429,6 +429,9 @@ def open_project(self, file_path, clear_thumbnails=True):
         app = get_app()
+        settings = app.get_settings()
+        # Set default load-project path to this path.
+        settings.setDefaultPath(settings.actionType.LOAD, os.path.dirname(file_path))
```
Ditto on file_path being reduced to folder name here - already handled in setDefaultPath()

jonoomph commented on src/windows/models/files_model.py:
```diff
@@ -350,10 +351,10 @@ def add_files(self, files, image_seq_details=None, quiet=False, prevent_image_se
+                prev_path = settings.getDefaultPath(settings.actionType.IMPORT)
 
-                prev_path = app.project.get("import_path")
                 if dir_path != prev_path:
```
I'm not sure we need this logic anymore: `if dir_path != prev_path`... since we are only updating a setting... and no longer updating the project.

jonoomph commented:
Some comments and a few suggestions. Otherwise looks good.

Commit:
- Set IMPORT when fixing broken paths
- Log reducing to directory
- Let setDefaultDirectory take care of reducing to directory
- Simplify logic in  addFiles

jonoomph commented:
LGTM
